effectful/ops/types.py

```diff
@@ -422,60 +423,123 @@ def __fvs_rule__(self, *args: Q.args, **kwargs: Q.kwargs) -> inspect.BoundArgume
 
         return result_sig
 
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self.__name__}, {self.__signature__})"
```

Comment:
Note I haven't changed `__repr__` or `__str__`, I just moved `__call__` down to be next to the modified `__init_subclass__` so the recursion there is easier to follow.